Resources/public/js/views/dashboard/ez-dashboardblockallcontentview.js

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-dashboardblockallcontentview', function (Y) {
+    'use strict';
+
+    /**
+     * Provides the All Content Dashboard Block View class
+     *
+     * @module ez-dashboardblockallcontentview
+     */
+    Y.namespace('eZ');
+
+    var BLOCK_IDENTIFIER = 'all-content',
+        CLASS_LOADING = 'is-loading',
+        CLASS_ROW_SELECTED = 'is-row-selected',
+        SELECTOR_ROW = '.ez-allcontent-block-row',
+        SELECTOR_CONTENT = '.ez-allcontent-block-content',
+        EVENTS = {};
+
+    EVENTS[SELECTOR_ROW] = {tap: '_uiShowRowOptions'};
+
+    /**
+     * The all content dashboard block view
+     *
+     * @namespace eZ
+     * @class DashboardBlockAllContentView
+     * @constructor
+     * @extends eZ.DashboardBlockBaseView
+     */
+    Y.eZ.DashboardBlockAllContentView = Y.Base.create('dashboardBlockAllContentView', Y.eZ.DashboardBlockBaseView, [Y.eZ.AsynchronousView], {
+        events: EVENTS,
+
+        initializer: function () {
+            this._fireMethod = this._getContentData;
+            this._watchAttribute = 'contentList';
+            this._clickOutsideHandler = null;
+
+            this._set('identifier', BLOCK_IDENTIFIER);
+            this.get('container').addClass(this._generateViewClassName(Y.eZ.DashboardBlockBaseView.NAME));
+        },
+
+        /**
+         * Renders the dashboard block view
+         *
+         * @method render
+         * @return {eZ.DashboardBlockAllContentView} the view itself
+         */
+        render: function () {
+            var rows = [],
+                methodName = 'addClass',
+                contentList = this.get('contentList');
+
+            if (contentList) {
+                methodName = 'removeClass';
+                rows = this._prepareRowsData(contentList);
```

Comment:
`rows` is a quite low level term. We are  actually displaying a list of Content items. Also I see in `_prepareRowsData` that for each Content item, you build a custom object. Instead, it would be better to build what we call a *content struct* and to benefit from the `toJSON` method in the models. So basically, each Content item should be represented by a *content struct* object in the template build like:
```js
{
    content: content.toJSON(),
    contentType: contentType.toJSON(),
}
```